### PR TITLE
Implement register firebase token v2

### PR DESCRIPTION
**Pull request by nucastcle**

### CODE REVIEW

File: apps/authentications/src/app/controllers/authentications.controller.v2.ts [hunk 1]
```diff
@@ -322,4 +324,23 @@ export class AuthenticationControllerV2 {
...
   }
+
+  @CastcleBasicAuth()
+  @Post('register/notification')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  registerToken(
+    @Auth() { account }: Authorizer,
+    @Body() body: RegisterFirebaseDto,
+  ) {
+    this.authenticationService.createAccountDevice(body, account);
```
Comment (kittipong-castcle) [inline apps/authentications/src/app/controllers/authentications.controller.v2.ts:335]:
```suggestion
    await this.authenticationService.createAccountDevice(body, account);
```

File: apps/authentications/src/app/controllers/authentications.controller.v2.ts [hunk 2]
```diff
@@ -322,4 +324,23 @@ export class AuthenticationControllerV2 {
...
+  }
+
+  @CastcleBasicAuth()
+  @Delete('register/notification')
+  deleteToken(
+    @Auth() { account }: Authorizer,
+    @Body() body: RegisterFirebaseDto,
+  ) {
+    this.authenticationService.deleteAccountDevice(body, account);
```
Comment (kittipong-castcle) [inline apps/authentications/src/app/controllers/authentications.controller.v2.ts:344]:
```suggestion
    await this.authenticationService.deleteAccountDevice(body, account);
```

File: libs/database/src/lib/services/authentication.service.v2.ts [hunk 3]
```diff
@@ -870,4 +871,60 @@ export class AuthenticationServiceV2 {
       activation.verifyToken,
     );
   }
+
+  async createAccountDevice(body: RegisterFirebaseDto, account: Account) {
+    const devices = account?.devices?.find(
```
Comment (kittipong-castcle) [inline libs/database/src/lib/services/authentication.service.v2.ts:876]:
```suggestion
    const device = account.devices?.find(
```
account is required

File: apps/authentications/src/app/controllers/authentications.controller.v2.ts [hunk 4]
```diff
@@ -322,4 +324,23 @@ export class AuthenticationControllerV2 {
...
+  @HttpCode(HttpStatus.NO_CONTENT)
+  registerToken(
+    @Auth() { account }: Authorizer,
+    @Body() body: RegisterFirebaseDto,
+  ) {
+    this.authenticationService.createAccountDevice(body, account);
+  }
+
+  @CastcleBasicAuth()
+  @Delete('register/notification')
```
Comment (kittipong-castcle) [inline apps/authentications/src/app/controllers/authentications.controller.v2.ts:339]:
```suggestion
  @Delete('register/notification')
  @HttpCode(HttpStatus.NO_CONTENT)
```

File: libs/database/src/lib/services/authentication.service.v2.ts [hunk 5]
```diff
@@ -870,4 +871,60 @@ export class AuthenticationServiceV2 {
...
+          $addToSet: {
+            devices: {
+              uuid: body.uuid,
+              platform: body.platform,
+              firebaseToken: body.firebaseToken,
+            },
+          },
+        },
+      );
+    }
```
Comment (kittipong-castcle) [inline libs/database/src/lib/services/authentication.service.v2.ts:909]:
```suggestion
    if (device) device.firebaseToken = body.firebaseToken;
    else (account.devices ||= []).push({
              uuid: body.uuid,
              platform: body.platform,
              firebaseToken: body.firebaseToken,
            })
            
            account.markModified('devices');
            await account.save();
```

**Review by kittipong-castcle — CHANGES_REQUESTED** [review]

**Review by kittipong-castcle — APPROVED** [review]

**Review by julapong-castcle — APPROVED** [review]

**Pull request merged**